### ISSUE TITLE
combat-trainer releasing SOV at start of hunt

### ISSUE THREAD

**Issue by drgabellia** [rpherbig/dr-scripts#3173]
I have SOV (Steps of Vuan, cyclic spell) set up in waggle, default, and also buff_spells settings.  It's the only place in my yaml where sov appears.
  sov: &sov
    Steps of Vuan:   #5/25
      abbrev: sov
      mana: 22
      cyclic: true

  default: &default
    << : *shadowling
    << : *wd
    << : *aus
    << : *sov
    << : *lw

buff_spells: &buff_spells
  << : *aus
  << : *maf
  << : *shadowling
  << : *wd
  << : *sov
  << : *lw

For some, inexplicable reason, at the beginning of each hunt, combat-trainer releases SOV  (log below) and it never casts it again.  (Even though it does cast maf which I have in buff_spells but not in default)

All default buffs have been cast...and this is the start of my hunt:

--- Lich: combat-trainer active.

[combat-trainer]>stance set 100 0 80

Setting your Evasion stance to 100%, your Parry stance to 0%, and your Shield stance to 80%.  You have 8 stance points left.

[combat-trainer]>get my arrowhead

You maintain control of the Steps of Vuan spell.

[combat-trainer]>drop arrowhead

You get a serrated-bodkin arrowhead from inside your leather pack.
You drop a serrated-bodkin arrowhead.

[combat-trainer]>tap Meraud orb

I could not find what you were referring to.

[combat-trainer]>rummage /B my duffel bag

You feel about a fine silk duffel bag.

[combat-trainer]>release SOV

The refractive field surrounding you fades away.

[combat-trainer]>rummage /B my duffel bag

Hopefully someone can figure out why combat-trainer is releasing this spell.  Are there any other MoonMages experiencing the same issue?

**Comment by Tarhish** [comment]
Crap, this was a weird one, but you had Just enough info in that log to go off of.  The 'rummage /b' comes from checking how many boxes you have for either a box hunt or just establishing how many you need for the box_loot_limit: to be reached.  When DRC (common.lic) tries to rummage in your pack for boxes it can't do that when you're invisible, so it drops any active spell tagged with invisibility: true (SOV is by default).  Does that sound right to you?

But you say it's not recasting it at all either?  That I can't figure out atm.  Got any more information about that?

**Comment by drgabellia** [comment]
I don't have the hunt/box_loot_limit set anywhere....as this toon doesn't pick up boxes.  As for it not casting again...makes no sense.  ALL her other spells listed in buff spells cast.  The only difference is that this spell is cyclic.  Does that factor in somehow?

Thanks for looking into it.  I hope it can be figured out....as she really needs this spell for her higher end hunts.

**Comment by drgabellia** [comment]
One other comment which may or may not pertain....while she doesn't pick up boxes, I do have - pick set in her before and after hunts as she keeps a pet supply in her duffel bag.  Could that be causing the issue?

**Comment by Tarhish** [comment]
I didn't see that until you mentioned it now, but yeah, pet picking in combat also checks the boxes you have on you in your pet storage, and unlike the others I think it does it every time you try to pick rather than just at the start.

So I guess the next question might be: are you pretty sure it's not recasting SOV at all in combat, or might it be casting it then dropping it 10-30 seconds later when it tries to pick pets?

**Comment by drgabellia** [comment]
She does NOT pick pets during combat.  - pick is set only at before/after.  However, that said, I did some further experimenting. 
#1 added refractive field (also an invisibility spell) to her default and buff_spells.
Combat-Trainer now releases SoV and rf at beginning of hunt.  However....it recasts rf during hunt and keeps it up entire hunt, while NOT recasting SoV.

#2 commented out - pick at beginning of hunt. 
Combat-Trainer does NOT release either SoV or rf.

So..we have two separate issues that I can see.  SoV is not being recast during hunt even though listed in buff_spells and - pick is causing combat-trainer to drop invisibility spells even though she is not picking during the hunt.

**Comment by Tarhish** [comment]
So I came back to look at this with fresh eyes.  There's actually not very many places in combat-trainer where it's possible for it to release a specific spell like SOV.  Most of them only occur on shutdown or for other hardcoded spells like ignite, so the only thing that jumps out at me is cyclic stuff.  If you're trying to cast a cyclic and you already have one being channeled then it releases the prior one immediately.  It's also interesting that RF isn't being released, so that part, at least, isn't about invisibility.

Do you have a cyclic spell like Starlight Sphere in your offensive_spells, which I can't see?

**Comment by drgabellia** [comment]
No cyclic spells running other than SOV.

offensive_spells:
- skill: Targeted Magic
  name: Telekinetic Storm
  abbrev: tks
  mana: 20
  cambrinth:
  - 30
  - 30
  min_threshold: 2
  slivers: false  
  harmless: true
#- name: Mental Blast
#  mana: 20
#  cambrinth:
#  - 18
#- skill: Debilitation
#  name: Calm
#  abbrev: calm
#  mana: 10
#  cambrinth:
#  - 20
#- name: Starlight Sphere
#  mana: 14
#  cast: cast spider
- skill: Targeted Magic  
  name: Partial Displacement
  abbrev: pd
  mana: 25
  cambrinth:
  - 25  
#- skill: Sorcery
#  name: Halt
#  abbrev: halt
#  mana: 13
#  cambrinth:
#  - 14
#  cast_only_to_train: true
#- skill: Sorcery
#  name: Stun Foe
#  abbrev: sf
#  mana: 12
#  cambrinth:
#  - 15
#  cast_only_to_train: true
#- skill: sorcery
#  name: Soul Sickness
#  abbrev: sick
#  mana: 5
#  cambrinth:
#  - 15
#  expire: The spiritual weight lifts off 
#- name: Dazzle
#  abbrev: dazzle
#  mana: 20
#  cambrinth:
#  - 13
- skill: Debilitation  
  name: Sleep
  abbrev: sleep
  mana: 5
  harmless: true
  prep_tims: 1
  cambrinth:
  - 10
  
offensive_spell_cycle:
#- Starlight Sphere
- Telekinetic Storm
#- Mental Blast
#- Calm
- Partial Displacement
#- Halt
#- Dazzle
- Sleep
#- Soul Sickness
#- Stun Foe

**Comment by drgabellia** [comment]
Yikes!!!!! I have no idea why that posted so large.  Sorry!

**Comment by Tarhish** [comment]
The formatting here has taken the dashes and hashes and turned them into dots and big letters, but if you add three tildes above and below the paste like this:

\`\`\`
YAML STUFF
\`\`\`

I disabled that, but if you do it that way it'll come out looking like this:
```
YAML STUFF
```

and it won't get messed up.  

Still, I see Starlight Sphere on that list, which is an offensive cyclic spell, but I can't tell if it's commented out or not?  If it isn't commented out then it'll drop SOV every time it prepares to cast that.

**Comment by drgabellia** [comment]
Starlight Sphere is commented out.  I have no cyclic spells running except for SOV.

**Comment by drgabellia** [comment]
I did a workaround.  I took it out of my default waggle, but left it in my buff_spells waggle.  This way, when I buff BEFORE a hunt, sov and rf are not called.  Once the hunt has started, they are called then.  I don't know why it kept releasing both spells immediately after the hunt started, (when in default waggle) and then would only cast rf and not sov after hunt started.....but my workaround is working, so I'm happy.  Hopefully this little bug will not annoy anyone else.  Thank you everyone for your suggestions, really appreciate it!
